src/collections/Table/Table.js

```diff
@@ -10,43 +10,97 @@ export default class Table extends Component {
     }),
   };
 
+  static defaultProps = {
+    selectedRows: [],
+    sort: {
+      key: null,
+      direction: 'descending',
+    },
+  }
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      selectedRows: this.props.selectedRows,
+    };
+  }
+
   static getSafeCellContents(content) {
     // React cannot render objects, stringify them instead
     return _.isObject(content) ? JSON.stringify(content) : content;
   }
 
+  _isRowSelected(index) {
+    return _.includes(this.state.selectedRows, index);
+  }
+
+  _isSelectable = () => {
+    return _.includes(this.props.className, 'selectable');
+  };
+
+  _unselectRow(index) {
+    if (!this._isSelectable()) return;
+    this.setState({
+      selectedRows: _.without(this.state.selectedRows, index)
+    });
+  }
+
+  _selectRow(index) {
+    if (!this._isSelectable()) return;
+    this.setState({selectedRows: [index]});
+  }
+
+  _unselectAllRows() {
+    this.setState({selectedRows: []});
+  }
+
+  _toggleSelectRow(index) {
+    if (this._isRowSelected(index)) {
+      this._unselectRow(index);
+    } else {
+      this._selectRow(index);
+    }
+  }
+
+  _handleSelectRow(rowItem, rowIndex) {
+    this._toggleSelectRow(rowIndex);
+    if (this.props.onSelectRow) this.props.onSelectRow(rowItem, rowIndex);
+  }
+
   _handleSortHeaderChange(key, direction) {
     const {onSortChange} = this.props;
     if (onSortChange) {
+      this._unselectAllRows();
       onSortChange(key, direction);
     }
   }
 
   _getHeaders() {
-    return Children.map(this.props.children, (column) => {
-      const key = column.props.dataKey;
-      const content = column.props.headerRenderer
-        ? column.props.headerRenderer(this.props.data[0])
-        : _.startCase(column.props.dataKey);
-      const sorted = _.get(this, 'props.sort.key') === key;
-      const direction = _.get(this, 'props.sort.direction', 'descending');
+    const {children, data, sort} = this.props;
+
+    return Children.map(children, (column) => {
+      const {dataKey, headerRenderer} = column.props;
+      const content = headerRenderer ? headerRenderer(data[0]) : _.startCase(dataKey);
+      const isSorted = sort.key === dataKey;
       const onClick = () => this._handleSortHeaderChange(
-        key, direction === 'ascending' ? 'descending' : 'ascending'
+        dataKey, sort.direction === 'ascending' ? 'descending' : 'ascending'
       );
       const classes = classNames('sd-table-header', {
-        sorted,
-        ascending: sorted && direction === 'ascending',
-        descending: sorted && direction === 'descending',
+        sorted: isSorted,
+        ascending: isSorted && sort.direction === 'ascending',
+        descending: isSorted && sort.direction === 'descending',
       });
 
-      return <th className={classes} key={key} onClick={onClick}>{content}</th>;
+      return <th className={classes} key={dataKey} onClick={onClick}>{content}</th>;
```

Comment:
No functional changes to this method.  Just destructuring, use of the defaultProp values, and variable name updates.